expression/builtin_time_vec.go

```diff
@@ -995,11 +995,60 @@ func (b *builtinTimeToSecSig) vecEvalInt(input *chunk.Chunk, result *chunk.Colum
 }
 
 func (b *builtinStrToDateDatetimeSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinStrToDateDatetimeSig) vecEvalTime(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	dateBuf, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(dateBuf)
+	if err = b.args[0].VecEvalString(b.ctx, input, dateBuf); err != nil {
+		return err
+	}
+
+	formatBuf, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(formatBuf)
+	if err = b.args[1].VecEvalString(b.ctx, input, formatBuf); err != nil {
+		return err
+	}
+
+	result.ResizeTime(n, false)
+	result.MergeNulls(dateBuf)
+	result.MergeNulls(formatBuf)
+	times := result.Times()
```

Comment:
The `times`  slice need initialize according to `types.ZeroDatetime`

Reply:
Could you elaborate on this? Checked all vectorized implementations where `result.Times()` is used and could not find any addition initializations there.

Reply:
Maybe I'm wrong. wshwsh12 had suggested to me that the Time slice need initialize according to types.ZeroDatetime. So I suggested you do so.
This is the link about that. #12825 (comment)
In my opinion, the zero-value of Time is not a ZeroDateTime value. so we need to initialization them.
I'm confounded that I could not find any Times slice initializations also.

Reply:
The initialization is only needed when we have cases `continue` but not `NULL` cases in the fop loop.

```go
<the initialization code>
for i := 0; i < n; i++ {
    if xxx {
        // we should initialize before the for loop since here we continue but not set ` times[i]`
        continue
    }
    times[i] = t
}
```

Another alternative implementation is making sure to set `times[i]` carefully in each for loop.

```go
// need not initialize here.
for i := 0; i < n; i++ {
    if xxx {
        times[i] = types.ZeroDatetime
        continue
    }
    times[i] = t
}
```

Reply:
Thanks for clarification, guys.

I used the second way (`SetNull/continue`) and, if I understand you correctly, we don't need any initializations here, right?

Reply:
Yes.